src/vm/gdbjit.cpp

```diff
@@ -964,6 +975,56 @@ TypeKey* TypeInfoBase::GetTypeKey()
     return &typeKey;
 }
 
+void TypeDefInfo::DumpStrings(char *ptr, int &offset)
+{
+    if (ptr != nullptr)
+    {
+        strcpy(ptr + offset, m_typedef_name);
+        m_typedef_name_offset = offset;
+    }
+    offset += strlen(m_typedef_name) + 1;
+}
+
+void TypeDefInfo::DumpDebugInfo(char *ptr, int &offset)
+{
+    if (ptr != nullptr)
+    {
+        DebugInfoTypeDef buf;
+        buf.m_typedef_abbrev = 3;
+        buf.m_typedef_name = m_typedef_name_offset;
+        buf.m_typedef_type = offset + sizeof(DebugInfoTypeDef);
+        m_typedef_type_offset = offset;
+
+        memcpy(ptr + offset,
+               &buf,
+               sizeof(DebugInfoTypeDef));
+    }
+
+    offset += sizeof(DebugInfoTypeDef);
+}
+
+void ByteTypeInfo::DumpStrings(char* ptr, int& offset)
+{
+    PrimitiveTypeInfo::DumpStrings(ptr, offset);
+    m_typedef_info->m_typedef_name = new (nothrow) char[strlen(m_type_name) + 1];
+    if (strcmp(m_type_name, "System.Byte") == 0)
+        strcpy(m_typedef_info->m_typedef_name, "byte");
+    else if (strcmp(m_type_name, "System.SByte") == 0)
+        strcpy(m_typedef_info->m_typedef_name, "sbyte");
+    else if (strcmp(m_type_name, "char16_t") == 0)
```

Comment:
Yes, you're right. We use `char16_t` to show unicode characters in lldb correctly. You might see that we renamed `System.Char` to `char16_t` here  [gdbjit.cpp:205](https://github.com/lucenticus/coreclr/blob/a6266a598d3893bb76acbb544ff092ea3da8363a/src/vm/gdbjit.cpp#L205). These hacks are needed because DWARF standard doesn't support C# language, and lldb doesn't support C# type system.